codegen/src/api/storage.rs

```diff
@@ -41,22 +42,19 @@ pub fn generate_storage(
     pallet: &PalletMetadata<PortableForm>,
     types_mod_ident: &syn::Ident,
     crate_path: &CratePath,
-) -> TokenStream2 {
-    let storage = if let Some(ref storage) = pallet.storage {
-        storage
-    } else {
-        return quote!()
+) -> Result<TokenStream2, CodegenError> {
+    let Some(storage) = &pallet.storage else {
+        return Ok(quote!())
     };
 
-    let storage_fns: Vec<_> = storage
-        .entries
-        .iter()
-        .map(|entry| {
-            generate_storage_entry_fns(metadata, type_gen, pallet, entry, crate_path)
-        })
-        .collect();
+    let mut storage_fns = Vec::new();
+    for entry in &storage.entries {
+        let storage_fn =
+            generate_storage_entry_fns(metadata, type_gen, pallet, entry, crate_path)?;
+        storage_fns.push(storage_fn);
+    }
```

Comment:
I guess some of this could have been done with a `.collect::<Result<Vec<_>,_>()` style thing instead of changing to loops, but I'm happy with either really!

Reply:
Have reverted to using `collect::<...>` everywhere, that sounds pretty cool!

One strange thing about it, the compiler seems to be quite strict in resolving `collect::<Result<Vec<_>, _>`.
If the Error is not explicitly typed in the closure, then the scope in which the compiler finds Error type is not increased. Such is the case where we use `?` and the error could be deduced from the function's return type.
This leads to some places where the `CodegenError` needs to be typed next to collect.